### ISSUE TITLE
Multisite: Deleted sites show as "phantom" sites on Jetpack -> Sites

### ISSUE THREAD

**Issue by richardmtl** [Automattic/jetpack#6315]
I recently deleted several subsites in my multisite installation. They are marked as "Deleted" if I go to http://domain.tld/wp-admin/network/site-info.php?id=XX :

<img width="342" alt="screen shot 2017-02-07 at 2 56 20 pm" src="https://cloud.githubusercontent.com/assets/3676359/22708793/a0f1af7a-ed45-11e6-8cb1-b26e4ddf9a24.png">

But on http://domain.tld/wp-admin/network/admin.php?page=jetpack I get this:

<img width="816" alt="screen shot 2017-02-07 at 14 57 45" src="https://cloud.githubusercontent.com/assets/3676359/22708849/d6c564e8-ed45-11e6-98f6-af7229883e94.png">

Now, this might very well be limited to my own multisite. I recently switched host, and cleaned up a lot of cruft while I was at it, manually deleting tables and such for many subsites. So, ideally someone would test with another multisite network to see if they can reproduce this.

**Comment by rcoll** [comment]
@richardmtl I do not get the same results when testing. Are you using a persistent object cache?

**Comment by richardmtl** [comment]
@rcoll No, I am not. If this is not reproducible, then I'll just chalk it up to wonkiness due to my manually editing some tables, and changing hosts.

**Comment by jeherve** [comment]
@richardmtl We actually just made some changes that might help. Could you try to apply this patch and let me know if it helps?
https://github.com/Automattic/jetpack/pull/3405/files

**Comment by richardmtl** [comment]
No, that didn't fix it for me @jeherve

**Comment by richardmtl** [comment]
I've messed around with my site and db (moved hosts a couple of times, manually edited the db). That being said, I still have this issue. Might be edge-case enough to merit a wontfix though; I'll let someone else decide that!
